### PR TITLE
Add the interface of reading the crosstalk map for ALLEGRO ECAL barrel

### PR DESCRIPTION
**Pull request by zwu0922**
BEGINRELEASENOTES
- Added ICaloReadCrosstalkMap interface definition to support the service tool that reads the crosstalk map of the ALLEGRO ECAL barrel

ENDRELEASENOTES

### CODE REVIEW

File: k4Interface/include/k4Interface/ICaloReadCrosstalkMap.h [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ *
+ *  @author Zhibo Wu
+ */
+
+class ICaloReadCrosstalkMap : virtual public IAlgTool {
+public:
+  DeclareInterfaceID(ICaloReadCrosstalkMap, 1, 0);
+
+  virtual std::vector<uint64_t>& getNeighbours(uint64_t aCellId) = 0;
+  virtual std::vector<double>& getCrosstalks(uint64_t aCellId) = 0;
```
Comment (andresailer) [inline k4Interface/include/k4Interface/ICaloReadCrosstalkMap.h:37]:
```suggestion
  virtual std::vector<uint64_t> const& getNeighbours(uint64_t aCellId) = 0;
  virtual std::vector<double> const& getCrosstalks(uint64_t aCellId) = 0;
```

File: k4Interface/include/k4Interface/ICaloReadCrosstalkMap.h [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ *
+ *  @author Zhibo Wu
+ */
+
+class ICaloReadCrosstalkMap : virtual public IAlgTool {
+public:
+  DeclareInterfaceID(ICaloReadCrosstalkMap, 1, 0);
+
+  virtual std::vector<uint64_t> const& getNeighbours(uint64_t aCellId) = 0;
+  virtual std::vector<double> const& getCrosstalks(uint64_t aCellId) = 0;
```
Comment (jmcarcell) [inline k4Interface/include/k4Interface/ICaloReadCrosstalkMap.h:37]:
```suggestion
  virtual std::vector<uint64_t> const& getNeighbours(uint64_t cellID) = 0;
  virtual std::vector<double> const& getCrosstalks(uint64_t cellID) = 0;
```
I think this is the convention I see more often

Comment (zwu0922) [reply k4Interface/include/k4Interface/ICaloReadCrosstalkMap.h]: Thanks. I have changed these two lines to follow the convention.

File: k4Interface/include/k4Interface/ICaloReadCrosstalkMap.h [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ *
+ *  @author Zhibo Wu
+ */
+
+class ICaloReadCrosstalkMap : virtual public IAlgTool {
+public:
+  DeclareInterfaceID(ICaloReadCrosstalkMap, 1, 0);
+
+  virtual std::vector<uint64_t> const& getNeighbours(uint64_t CellId) = 0;
+  virtual std::vector<double> const& getCrosstalks(uint64_t CellId) = 0;
```
Comment (jmcarcell) [inline k4Interface/include/k4Interface/ICaloReadCrosstalkMap.h:37]:
```suggestion
  virtual std::vector<uint64_t> const& getNeighbours(uint64_t cellID) = 0;
  virtual std::vector<double> const& getCrosstalks(uint64_t cellID) = 0;
```
I don't want to be pedantic but while we're at it the change I proposed is `cellID` not `CellId`. This is what DD4hep seems to be using in may places, see https://github.com/AIDASoft/DD4hep/blob/master/install/include/DD4hep/PolarGridRPhi2.h#L69

File: k4Interface/include/k4Interface/ICaloReadCrosstalkMap.h (hunk 2 above)
Comment (zwu0922) [reply k4Interface/include/k4Interface/ICaloReadCrosstalkMap.h]: Sorry. This time I used "Commit suggestion" to update the code

**Review by jmcarcell — APPROVED** [review]

**Pull request merged**